pkg/scheduler/factory/factory_test.go

```diff
@@ -283,24 +238,31 @@ func TestDefaultErrorFunc(t *testing.T) {
 		ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
 		Spec:       apitesting.V1DeepEqualSafePodSpec(),
 	}
-	handler := utiltesting.FakeHandler{
-		StatusCode:   200,
-		ResponseBody: runtime.EncodeOrDie(schedulertesting.Test.Codec(), testPod),
-		T:            t,
-	}
-	mux := http.NewServeMux()
+	client := fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testPod}})
+	requestReceived := false
+	client.PrependReactor("get", string(v1.ResourcePods),
```

Comment:
This seems like new behavior in the test. What problem does it solve, and is this test flaky? (I notice the 10ms sleep that's probably there for this reason.)

If we're asserting very specific interactions with the k8s API, then using reactors might be the way to go. I've found that it's often sufficient to inject objects and check state through the Clientset API itself. For example, https://github.com/kubernetes/client-go/tree/master/examples/fake-client

Reply:
@misterikkit It's not a new behaviour, I just want to use reactor to validate that if the request is received or the request is expected, like what the original test does using `ValidateRequest`.http.ValidateRequest. I found no other method to do the validation in a simple way.

Reply:
Ah, I see that now. What do you think about calling `client.Actions()` and doing checks on that slice, rather than using the reactor interface? I think that code would be a little simpler for others to understand.

Reply:
@misterikkit `client.Actions()` would be a better way, thanks.